### PR TITLE
Solution to challenges 0-4 (python)

### PR DESCRIPTION
**Pull request by Atakyn**

### CODE REVIEW

File: challenge_3/python/Atakyn/c3.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# find majority element in array
+
+
+def find_majority_element(array):
+    majority_value = len(array) // 2
+    values = {}
+    for e in array:
+        try:
+            values[e] += 1
```
Comment (moonpatroller) [inline challenge_3/python/Atakyn/c3.py]: Why not use defaultdict?

File: challenge_4/python/Atakyn/c4.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+def reverse(root):  # solution, accounts for unpaired nodes (e.g. if 10 were added to the challenge example tree)
+    if root is not None:
+        if root.left is not None:
+            temp = root.left
+            root.left = root.right
+            root.right = temp
+            reverse(root.left)
+            reverse(root.right)
+        elif root.right is not None:
```
Comment (moonpatroller) [inline challenge_4/python/Atakyn/c4.py]: I think this should be an if, not elif.

File: challenge_3/python/Atakyn/c3.py (hunk 1 above)
Comment (Atakyn) [inline challenge_3/python/Atakyn/c3.py]: Didn't know about that - switching to using it now

File: challenge_4/python/Atakyn/c4.py (hunk 2 above)
Comment (Atakyn) [inline challenge_4/python/Atakyn/c4.py]: Fair enough - fixed

**Comment by MJUIUC** [comment]
Looks good. Awesome job reviewing and writing!

**Pull request merged**